hazelcast/src/test/java/example/serialization/ReflectiveMainDTO.java

```diff
@@ -0,0 +1,235 @@
+/*
+ * Copyright (c) 2008-2022, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package example.serialization;
+
+import java.math.BigDecimal;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.OffsetDateTime;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+/**
+ * A class that includes all fields in the {@link MainDTO} class and some extra that are only
+ * supported by the reflective serializer:
+ * - Enum
+ * - List
+ * - Map
+ * - Set
+ * Used by {@link com.hazelcast.nio.serialization.compatibility.BinaryCompatibilityTest}.
+ */
+@SuppressWarnings({"checkstyle:ParameterNumber"})
+public class ReflectiveMainDTO {
+
+    public byte b;
+    public boolean bool;
+    public char c;
```

Comment:
this and `Character nullableC;` also extra fields

Reply:
I did not get it.

Reply:
sorry, I editted the comment. basically, it would make sense to put them after the extra fields comment below as well

Reply:
Done.